File: DSAlgorithms/Leetcode/c++/arrays_hash/1963-swaps_to_balance_string.cpp
class Solution {
public:
    int minSwaps(string s) {
       int close = 0;
       int max_close = 0;
       for(int i = 0; i < s.size(); i++) 
       {
           if(s[i] == '[')
                close -= 1;
            else
                close += 1;
            max_close = std::max(close, max_close);
       }
       return (max_close + 1)/2;
    }
};

File: dsa_algo/leetcode/ts/src/arrays_hash/1963-swaps_to_balance_string.ts
const minSwaps = (s: string): number =>{
    let closed: number = 0;
    let max_closed: number = 0;
    for(let i = 0; i < s.length; i++)
    {
        if(s[i] === "[")
            closed -= 1;
        else
            closed += 1;
        
        max_closed = Math.max(closed, max_closed);
    }
    return Math.floor((max_closed + 1)/2);

};